File: 1-100/Task03.js
// [3. 无重复字符的最长子串 - 力扣（LeetCode）](https://leetcode.cn/problems/longest-substring-without-repeating-characters/)
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  let left = 0;
  let right = 0;
  let max = 0;
  let arr = "";
  while (left <= right && right < s.length) {
    arr = s.slice(left, right);
    if (arr.indexOf(s[right]) == -1) {
      max = arr.length + 1 > max ? arr.length + 1 : max;
      right++;
    } else {
      max = arr.length > max ? arr.length : max;
      left = arr.indexOf(s[right]) + 1 + left;
    }
  }

  return max;
};

console.log(lengthOfLongestSubstring(" "));

File: leetcode/Task03.py
# Longest Substring Without Repeating Characters - LeetCode
# https://leetcode.com/problems/longest-substring-without-repeating-characters/Longest Substring Without Repeating Characters - LeetCode
class Solution:
    max_length: int = 0
    curr_str = ""

    def lengthOfLongestSubstring(self, s: str) -> int:
        left_ptr = 0
        right_ptr = 0
        while right_ptr < len(s) - 1:
            self.curr_str: str = s[left_ptr : right_ptr + 1]
            new_char: str = s[right_ptr + 1]
            index: int = self.curr_str.find(new_char)
            right_ptr = right_ptr + 1
            if index != -1:
                left_ptr = left_ptr + index + 1
            self.curr_str: str = s[left_ptr : right_ptr + 1]
            self.max_length = (
                self.max_length
                if len(self.curr_str) <= self.max_length
                else len(self.curr_str)
            )
        if len(s) == 1:
            self.max_length = 1
        return self.max_length


test = Solution()
print(test.lengthOfLongestSubstring(" "))
